packages/app/cypress/e2e/create-from-component.cy.ts

```diff
@@ -0,0 +1,189 @@
+      cy.get('@ComponentCard')
+      .within(() => {
+        cy.findByRole('button', {
+          name: 'Create from component',
+        }).should('be.visible')
+        .and('not.be.disabled')
+      })
+    })
+
+    it('Can be closed with the x button', () => {
+      cy.get('@ComponentCard').click()
+
+      cy.findByRole('button', { name: 'Close' }).as('DialogCloseButton')
+
+      cy.get('@DialogCloseButton').click()
+      cy.findByRole('dialog', {
+        name: 'Choose a component',
+      }).should('not.exist')
+    })
+
+    it('Lists Vue components in the project', () => {
+      cy.get('@ComponentCard').click()
+
+      cy.findByText('2 Matches').should('be.visible')
+
+      cy.findByText('App').should('be.visible')
+      cy.findByText('HelloWorld').should('be.visible')
+    })
+
+    it('Allows for the user to search through their components', () => {
+      cy.get('@ComponentCard').click()
+
+      cy.findByText('*.vue').should('be.visible')
+      cy.findByText('2 Matches').should('be.visible')
+      cy.findByLabelText('file-name-input').type('HelloWorld')
+
+      cy.findByText('HelloWorld').should('be.visible')
+      cy.findByText('1 of 2 Matches').should('be.visible')
+      cy.findByText('App').should('not.exist')
+    })
+
+    it('shows success modal when component spec is created', () => {
+      cy.get('@ComponentCard').click()
+
+      cy.findByText('HelloWorld').should('be.visible').click()
+
+      cy.findByRole('dialog', {
+        name: defaultMessages.createSpec.successPage.header,
+      }).as('SuccessDialog').within(() => {
+        cy.contains(getPathForPlatform('src/components/HelloWorld.cy.js')).should('be.visible')
+        cy.findByRole('button', { name: 'Close' }).should('be.visible')
+
+        cy.findByRole('link', { name: 'Okay, run the spec' })
+        .should('have.attr', 'href', `#/specs/runner?file=src/components/HelloWorld.cy.js`)
+
+        cy.findByRole('button', { name: 'Create another spec' }).click()
+      })
+
+      // 'Create from component' card appears again when the user selects "create another spec"
+      cy.findByText('Create from component').should('be.visible')
+    })
+
+    it('runs generated spec', () => {
+      cy.get('@ComponentCard').click()
+
+      cy.findByText('HelloWorld').should('be.visible').click()
+
+      cy.findByRole('dialog', {
+        name: defaultMessages.createSpec.successPage.header,
+      }).as('SuccessDialog').within(() => {
+        cy.contains(getPathForPlatform('src/components/HelloWorld.cy.js')).should('be.visible')
+        cy.findByRole('button', { name: 'Close' }).should('be.visible')
+
+        cy.findByRole('link', { name: 'Okay, run the spec' })
+        .should('have.attr', 'href', `#/specs/runner?file=src/components/HelloWorld.cy.js`).click()
+      })
+
+      cy.findByText('<HelloWorld ... />').should('be.visible')
+    })
+  })
+
+  context('project with custom spec pattern', () => {
```

Comment:
This feels like we're duplicating the same tests, just with a different specPattern arg. Could we extract them to a reusable function or within a loop so they can be run with variable setup/args?

```
const validateComponentCard = (configFile: string, expectedLocation: string) => {
  beforeEach(() => {
      cy.scaffoldProject('no-specs-vue-2')
      cy.openProject('no-specs-vue-2', ['--config-file', configFile])
      cy.startAppServer('component')
      cy.visitApp()

      cy.findAllByTestId('card').eq(0).as('ComponentCard')
    })

    it('Shows create from component card for Vue projects with default spec patterns', () => {
...
}

context('project with default spec pattern', () => {
  validateComponentCard('configFile1', 'my-awesome-location-1')
})

context('project with custom spec pattern', () => {
  validateComponentCard('configFile2', 'my-awesome-location-2')
})
```